### DISCUSSION TITLE
Using metadata values to fetch license

### DISCUSSION THREAD

**Discussion by Ryan365Apps** [keygen-sh/community#71] [Q&A] [answered]
The docs show this line
https://api.keygen.sh/v1/accounts/{ACCOUNT}/licenses?metadata[customerId]=cust_af9d94bf5ad4

Here is my license Metadata
{
  "tid": "abcdef1-1234-1234-1234-abcdef1",
  "something": "abcdef1"
}

So my assumption is this syntax should work to get this specific license.

https://api.keygen.sh/v1/accounts/{ACCOUNT}/licenses?limit=1&metadata[tid]=abcdef1-1234-1234-1234-abcdef1

But I get an empty array.

Does anyone know what I'm missing?

**Comment by ezekg** [comment]
How are you authenticating? I'm assuming you're using a product token, so does this license belong to that particular product?

**Comment by ezekg** [comment, marked as answer]
Actually, looking at your request logs, you seem to have an extra newline character tagged onto the end of these queries:

```
GET /v1/accounts/<account>/licenses?limit=1&metadata[tid]=xxxx-xxxx-xxxx-xxxx-77895f82c2e5%0A
```

Note the `%0A` at the end, which is a newline. You'll need to remove that.

**Reply by taus** [reply]
~~is this search feature the same referenced here: https://keygen.sh/docs/search/ In particular i am interested to know if searching metadata with get parameters is what is refered to in the disclaimer about the underlying search api~~

Nevermind. I got here in a round about way. For anyone else wondering; this is the documentation that is refered to: https://keygen.sh/docs/api/licenses/#licenses-list-query-metadata

It is a documented and public feature :)
